projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentServiceV2.java

```diff
@@ -0,0 +1,176 @@
+ */
+@Service
+@RequiredArgsConstructor
+public class DeploymentServiceV2 {
+  private static final Logger log = LoggerFactory.getLogger(DeploymentServiceV2.class);
+
+  private final DockerRegistryService dockerRegistryService;
+  private final DeploymentProgress deploymentProgress;
+  private final JobImageBuilder jobImageBuilder;
+  private final JobImageDeployerV2 jobImageDeployer;
+  private final JobsRepository jobsRepository;
+  private final SupportedPythonVersions supportedPythonVersions;
+
+  /**
+   * Updates or creates a Kubernetes CronJob based on the provided configuration.
+   *
+   * <p>This method takes a CronJob configuration and checks if a CronJob with the same name already
+   * exists in the Kubernetes cluster. If the CronJob exists, it will be updated with the new
+   * configuration; otherwise, a new CronJob will be created. The method ensures that the CronJob in
+   * the cluster matches the desired state specified in the configuration.
+   *
+   * @param dataJob The data job to update or create.
+   * @param sendNotification if it is true the method will send a notification to the end user.
+   * @param cronJobs list of actual Kubernetes CronJobs.
+   */
+  public void updateDeployment(DataJob dataJob, Boolean sendNotification, Set<String> cronJobs) {
+    DataJobDeployment dataJobDeployment = dataJob.getDataJobDeployment();
+
+    if (dataJobDeployment == null) {
+      log.debug(
+          "Skipping the data job [job_name={}] deployment due to the missing deployment data",
+          dataJob.getName());
+      return;
+    }
+
+    // if (dataJobDeployment.getFailedDeployments() >= MAX_ATTEMPTS){
+    //  log.info("Skipping the data job [job_name={}] deployment due to the reach of max failed
+    // attempts in the latest attempt [git_commit_sha={}]", dataJob.getName(),
+    // dataJobDeployment.getLatestAttemptDeployedGitCommitSha());
+    //  return;
+    // }
+
+    // if (dataJob.getLatestJobDeploymentStatus().equals(DeploymentStatus.USER_ERROR) &&
+    //        dataJobDeployment.getLatestAttemptDeployedGitCommitSha() != null &&
+    //
+    // dataJobDeployment.getLatestAttemptDeployedGitCommitSha().equals(dataJobDeployment.getLatestFailedDeployedGitCommitSha())) {
+    //  log.debug("Skipping the data job [job_name={}] deployment due to the USER_ERROR in the
+    // latest attempt [git_commit_sha={}]", dataJob.getName(),
+    // dataJobDeployment.getLatestAttemptDeployedGitCommitSha());
+    //  return;
+    // }
+
+    JobDeployment jobDeployment = new JobDeployment();
+    jobDeployment.setDataJobTeam(dataJob.getJobConfig().getTeam());
+    jobDeployment.setDataJobName(dataJob.getName());
+
+    jobDeployment.setEnabled(dataJobDeployment.getEnabled());
+
+    DataJobResources dataJobResources = new DataJobResources();
+    dataJobResources.setCpuLimit(dataJobDeployment.getResourcesCpuLimit());
+    dataJobResources.setCpuRequest(dataJobDeployment.getResourcesCpuRequest());
```

Comment:
There was DeploymentModelConverter who was supposed to convert from one layer object to another.